src/turing_machine/machine.py

```diff
@@ -1,41 +1,83 @@
-class TuringMachine:
-    def __init__(self, tape, initial_state, final_states, transition_function):
+from abc import ABC, abstractmethod
+
+class Tape:
+    def __init__(self, tape):
         self.tape = list(tape)
         self.head = 0
-        self.state = initial_state
+
+    def move_left(self):
+        self.head -= 1
+
+    def move_right(self):
+        self.head += 1
+
+    def get_value(self):
+        return self.tape[self.head]
+
+    def set_value(self, val):
+        self.tape[self.head] = val
+
+    def __str__(self):
+        return ''.join(self.tape)
+
+
+class TuringMachine(ABC):
+    def __init__(self, tapes, initial_state, final_states):
+        self.initial_tapes = [Tape(tape) for tape in tapes]
+        self.initial_state = initial_state
         self.final_states = final_states
-        self.transition_function = transition_function
+        self.reset()
+
+    def reset(self):
+        self.tapes = [Tape(tape.tape) for tape in self.initial_tapes]
+        self.state = self.initial_state
         self.match_position = -1
 
+    def get_tapes(self):
+        return [tape.tape for tape in self.tapes]
+
+    @abstractmethod
+    def run_auto(self):
```

Comment:
Ta metoda powinna nazywać się `run_state`. Za to metoda `run_auto` powinna wykonywać to samo co `run_tick`, ale bez wywołania `input(..)` oraz `print_status()`, czyli powinna wykonać cały proces maszyny 'po cichu' i zakończyć działanie kiedy dotrzemy do stanu końcowego.